src/app/Http/Middleware/BackpackAdminGuard.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Backpack\Base\app\Http\Middleware;
+
+use Closure;
+
+class BackpackAdminGuard
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Closure                 $next
+     *
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if (config('backpack.base.separate_admin_session')) {
+            if (!\Auth::guard(config('backpack.base.admin_guard.name'))->check()) {
+                if ($request->ajax() || $request->wantsJson()) {
```

Comment:
Why not: `$request->expectsJson()` unless we don't like its pjson clause.

That said, why couldn't someone access admin functionality via JSON?

Reply:
Hmm... this might be an issue. @OwenMelbz wouldn't this prevent ajax calls to pages that are behind the ```backpack_middleware()```, if the new guard is used?

Reply:
This only applies is they are using separate sessions - if they're using the default laravel guard it will never affect it as that will have already handled the permission - baring in mind this whole implementation was based around the idea of using the default laravel guard for everything - then if turned on it simply adds another condition into the chain for the backend